src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,123 @@
+     * blank file is created instead. Throws an exception if the program cannot create a new file or there is an
+     * error obtaining the data from the file path.
+     *
+     * @return A list of tasks loaded from the file path
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File file = new File(this.FILE_PATH);
+            Scanner scanner = new Scanner(file);
+
+            while (scanner.hasNextLine()) {
+                String data = scanner.nextLine();
+                Task task = fromData(data);
+                if (!task.isEmpty()) {
+                    tasks.add(task);
+                }
+            }
+
+            scanner.close();
+        } catch (IOException e) {
+            if (!(e instanceof FileNotFoundException)) {
+                throw new DukeException("Unexpected error when getting data.");
+            }
+            File file = new File(this.FILE_PATH);
+            try {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            } catch (IOException ex) {
+                throw new DukeException("Error when trying to create new file.");
+            }
+        }
+        return tasks;
+    }
+
+    /**
+     * Saves all the tasks into their corresponding string representation to be loaded in the future. Throws an
+     * exception if there is an error accessing or saving the data to the file path.
+     *
+     * @param taskList The list of tasks to be saved
+     * @throws DukeException If there is an error saving the data of the tasks
+     */
+    public void updateData(TaskList taskList) throws DukeException {
+        try {
+            FileWriter fileWriter = new FileWriter(this.FILE_PATH);
+            String data = toData(taskList);
+            fileWriter.write(data);
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new DukeException("Error has occurred when saving data.");
+        }
+    }
+
+    /**
+     * Converts the list of tasks provided to the corresponding string representation to be saved.
+     *
+     * @param taskList The list of tasks
+     * @return The string representation of all the tasks
+     */
+    private String toData(TaskList taskList) {
+        String output = "";
+        for (Task task : taskList.getTasks()) {
+            output += task.toData();
+            output += "\n";
+        }
+        return output;
+    }
+
+    /**
+     * Converts a string representation of the task into its respective task object and is returned.
+     *
+     * @param data The string representation of the task data
+     * @return A corresponding task to the provided data
+     * @throws DukeException If the string representation does not match any format of tasks
+     */
+    private Task fromData(String data) throws DukeException {
```

Comment:
Perhaps the method name can start with a verb, ie: readFromFile, so that it will comply with Java naming coding standard. I noticed the same issue in several other places too.

Reply:
Except for fromData -> readFromData, would toData also count